src/ops/controlflow.rs

```diff
@@ -190,43 +180,73 @@ impl OpTrait for BasicBlock {
     }
 
     fn dataflow_signature(&self) -> Option<FunctionType> {
-        Some(match self {
-            BasicBlock::DFB {
-                extension_delta, ..
-            } => FunctionType::new(type_row![], type_row![]).with_extension_delta(extension_delta),
-            BasicBlock::Exit { .. } => FunctionType::new(type_row![], type_row![]),
-        })
+        Some(
+            FunctionType::new(type_row![], type_row![]).with_extension_delta(&self.extension_delta),
+        )
     }
 
     fn non_df_port_count(&self, dir: Direction) -> usize {
-        match self {
-            Self::DFB { tuple_sum_rows, .. } if dir == Direction::Outgoing => tuple_sum_rows.len(),
-            Self::Exit { .. } if dir == Direction::Outgoing => 0,
-            _ => 1,
+        match dir {
+            Direction::Incoming => 1,
+            Direction::Outgoing => self.tuple_sum_rows.len(),
         }
     }
 }
 
-impl BasicBlock {
-    /// The input signature of the contained dataflow graph.
-    pub fn dataflow_input(&self) -> &TypeRow {
-        match self {
-            BasicBlock::DFB { inputs, .. } => inputs,
-            BasicBlock::Exit { cfg_outputs } => cfg_outputs,
+impl OpTrait for Exit {
+    fn description(&self) -> &str {
+        "A CFG exit block node"
+    }
+    /// Tag identifying the operation.
+    fn tag(&self) -> OpTag {
+        Self::TAG
+    }
+
+    fn other_input(&self) -> Option<EdgeKind> {
+        Some(EdgeKind::ControlFlow)
+    }
+
+    fn other_output(&self) -> Option<EdgeKind> {
+        Some(EdgeKind::ControlFlow)
+    }
+
+    fn dataflow_signature(&self) -> Option<FunctionType> {
+        Some(FunctionType::new(type_row![], type_row![]))
+    }
+
+    fn non_df_port_count(&self, dir: Direction) -> usize {
+        match dir {
+            Direction::Incoming => 1,
+            Direction::Outgoing => 0,
         }
     }
+}
 
+/// Functionality shared by DFB and Exit CFG block types.
+pub trait BasicBlock {
+    /// The input signature of the contained dataflow graph.
```

Comment:
The dataflow signature of the values input to the block? (Exit blocks do not have a "contained dataflow graph")